dashboard/src/actions/quisbyChartActions.js

```diff
@@ -0,0 +1,124 @@
+import * as TYPES from "./types.js";
+
+import { DANGER, ERROR_MSG } from "assets/constants/toastConstants";
+
+import API from "../utils/axiosInstance";
+import { showToast } from "./toastActions";
+import { uriTemplate } from "../utils/helper";
+
+const chartTitleMap = {
+  gb_sec: "Bandwidth",
+  trans_sec: "Transactions/second",
+  usec: "Latency",
+};
+
+export const getQuisbyData = (dataset) => async (dispatch, getState) => {
+  try {
+    dispatch({ type: TYPES.LOADING });
+    const endpoints = getState().apiEndpoint.endpoints;
+    dispatch({
+      type: TYPES.SET_ACTIVE_RESOURCEID,
+      payload: dataset.resource_id,
+    });
+    const response = await API.get(
+      uriTemplate(endpoints, "datasets_visualize", {
+        dataset: dataset.resource_id,
+      })
+    );
+    if (response.status === 200 && response.data.json_data) {
+      dispatch({
+        type: TYPES.GET_QUISBY_DATA,
+        payload: response.data.json_data,
+      });
+      dispatch({
+        type: TYPES.IS_UNSUPPORTED_TYPE,
+        payload: "",
+      });
+      dispatch(parseChartData());
+    }
+    dispatch({ type: TYPES.COMPLETED });
+  } catch (error) {
+    if (error?.response) {
+      const errorMsg = error.response.data.message;
+      const isUnsupportedType = errorMsg?.toLowerCase().includes("unsupported");
+      if (isUnsupportedType) {
+        dispatch({
+          type: TYPES.IS_UNSUPPORTED_TYPE,
+          payload: errorMsg,
+        });
+      }
+    } else {
+      dispatch(showToast(DANGER, ERROR_MSG));
+    }
```

Comment:
I'm confused about when we set `IS_UNSUPPORTED_TYPE`.  We  _don't_ set it on error, _except_ if the error has a message which includes `"unsupported"`?  (So, we don't set it if we execute line 51 and we don't execute it in all cases for other errors?)

Reply:
Updated the `if-clause`